great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py

```diff
@@ -366,13 +366,13 @@ def get_data_for_batch_identifiers(
                 execution_engine, table_name, **splitter_kwargs
             )
         else:
-            batch_identifiers_list: List[
-                dict
-            ] = self.get_data_for_batch_identifiers_for_non_date_part_splitters(
-                execution_engine,
-                table_name,
-                processed_splitter_method_name,
-                splitter_kwargs,
+            batch_identifiers_list = (
```

Comment:
Shouldn't we declare `batch_identifiers_list: List[dict]` above the start of the `if`/`else` block?

Reply:
I'm happy to do that if that's a pattern we prefer.
I try to make as little change as possible in these initial type coverage PRs so I generally only make changes to make the type-checker happy and avoid stylistic changes.

Reply:
I've pushed these changes here and elsewhere.